index.js

```diff
@@ -39,22 +48,29 @@ function *register() {
 app.use(jwt({secret: process.env.JWT_SECRET}));
 
 app.use(route.get('/checkIn', checkIn));
+app.use(route.get('/locations', locations));
 
 function *checkIn() {
-  this.validateQuery('lat')
-    .required('Latitude required');
+  this.validateQuery('locationId')
+    .required('Location id required')
+    .isUuid('v4', 'Location id must be a valid UUID v4');
 
   this.validateQuery('lon')
     .required('Longitude required');
 
   var uuid = this.state.user['uuid'];
-  var query = 'INSERT INTO check_ins ("location", "users.id") VALUES (ST_GeomFromText(\'POINT(' + this.vals.lat + ' ' + this.vals.lon + ')\', 4326), (SELECT id FROM users WHERE uuid=\'' + uuid + '\'));';
+  var query = 'INSERT INTO check_ins ("places.id", "users.id") VALUES (' + this.vals.locationId + ', (SELECT id FROM users WHERE uuid=\'' + uuid + '\'));';
   var result = yield this.pg.db.client.query_(query);
   this.body = '';
 }
 
+function *locations() {
+  result = yield this.pg.db.client.query_('SELECT uuid, name FROM places;');
+  this.body = result.rows;
+}
+
 function *distance() {
-  var query = 'SELECT location FROM check_ins WHERE ST_Distance_Sphere(location,ST_GeomFromText(\'POINT(59.3354419 18.0577941)\', 4326)) < 500;';
+  var query = 'SELECT coordinate FROM places WHERE ST_Distance_Sphere(coordinate, ST_GeomFromText(\'POINT(59.3354419 18.0577941)\', 4326)) < 500;';
```

Comment:
Line is too long.<br>'query' is defined but never used.<br>Too many errors. (94% scanned).